### PR TITLE
Added a new DataDumper renderer

### PR DESCRIPTION
**Pull request by autarch**

### CODE REVIEW

File: lib/Test2/Harness/Fact.pm [hunk 1]
```diff
@@ -242,7 +230,7 @@ True if the fact results in a failure. (Example: 'not ok')
 
 True if the fact adds to the test count.
 
-=item $plan_ref = $f->sets_plan()
```
Comment (exodist) [inline lib/Test2/Harness/Fact.pm]: What is the point of removing the parens from the docs here?

File: lib/Test2/Harness/Renderer/JSONDump.pm [hunk 2]
```diff
@@ -0,0 +1,122 @@
+package Test2::Harness::Renderer::JSONDump;
+use strict;
+use warnings;
+
+our $VERSION = '0.000012';
+
+use File::Basename qw/basename dirname/;
```
Comment (exodist) [inline lib/Test2/Harness/Renderer/JSONDump.pm]: Are any of these new deps non-core? Just curious, not a blocker or anything.

File: lib/Test2/Harness/Fact.pm (hunk 1 above)
Comment (autarch) [inline lib/Test2/Harness/Fact.pm]: Just trying to keep the docs consistent.

File: lib/Test2/Harness/Renderer/JSONDump.pm (hunk 2 above)
Comment (autarch) [inline lib/Test2/Harness/Renderer/JSONDump.pm]: Nope, all core. If non-core was okay it'd be nice to use Path::Tiny or something, but this works well enough.

**Comment by autarch** [comment]
This needs some more testing against real test suites. Please don't merge yet.

**Pull request closed without merging**